src/gt4py/next/program_processors/runners/dace_iterator/itir_to_sdfg.py

```diff
@@ -0,0 +1,355 @@
+                    if str(i_sh) in closure_sdfg.free_symbols:
+                        symbol_mapping[str(i_sh)] = o_sh
+
+            # Insert the closure's SDFG as a nested SDFG of the program.
+            nsdfg_node = last_state.add_nested_sdfg(
+                sdfg=closure_sdfg,
+                parent=None,
+                inputs=set(input_names) | set(connectivity_names),
+                outputs=set(output_names),
+                symbol_mapping=symbol_mapping,
+            )
+
+            # Connect the access nodes to the nested SDFG's inputs via edges.
+            for inner_name, access_node in zip(input_names, input_accesses):
+                memlet = create_memlet_full(access_node.data, program_sdfg.arrays[access_node.data])
+                last_state.add_edge(access_node, None, nsdfg_node, inner_name, memlet)
+
+            for inner_name, access_node in zip(connectivity_names, connectivity_accesses):
+                memlet = create_memlet_full(access_node.data, program_sdfg.arrays[access_node.data])
+                last_state.add_edge(access_node, None, nsdfg_node, inner_name, memlet)
+
+            for inner_name, access_node in zip(output_names, output_accesses):
+                memlet = create_memlet_full(access_node.data, program_sdfg.arrays[access_node.data])
+                last_state.add_edge(nsdfg_node, inner_name, access_node, None, memlet)
+
+        program_sdfg.validate()
+        return program_sdfg
+
+    def visit_StencilClosure(
+        self, node: itir.StencilClosure, array_table: dict[str, dace.data.Array]
+    ) -> dace.SDFG:
+        assert ItirToSDFG._check_no_lifts(node)
+        assert ItirToSDFG._check_shift_offsets_are_literals(node)
+        assert isinstance(node.output, itir.SymRef)
+
+        neighbor_tables = filter_neighbor_tables(self.offset_provider)
+        input_names = [str(inp.id) for inp in node.inputs]
+        conn_names = [connectivity_identifier(offset) for offset, _ in neighbor_tables]
+        output_names = [str(node.output.id)]
+
+        # Create the closure's nested SDFG and single state.
+        closure_sdfg = dace.SDFG(name="closure")
+        closure_state = closure_sdfg.add_state("closure_entry")
+
+        # Add DaCe arrays for inputs, output and connectivities to closure SDFG.
+        for name in [*input_names, *conn_names, *output_names]:
+            assert name not in closure_sdfg.arrays or (name in input_names and name in output_names)
+            if name not in closure_sdfg.arrays:
+                closure_sdfg.add_array(
+                    name,
+                    shape=array_table[name].shape,
+                    strides=array_table[name].strides,
+                    dtype=array_table[name].dtype,
+                )
+
+        # Get output domain of the closure
+        program_arg_syms: dict[str, ValueExpr | IteratorExpr] = {}
+        for name, type_ in self.storages.items():
+            if isinstance(type_, ts.ScalarType):
```

Comment:
For inputs of scalar type we are assuming that the value will be read from a symbol and stored into a container. If this is right, shouldn't this assignment take place in an entry state, before the main closure state?

Reply:
They could be, but there is no need to - the dataflow will determine the dependencies. 

I would rather argue that we don't need to make them scalars in the first place, since they can indeed be const. Let's leave it at what it is for now though.